### PR TITLE
Add docs about testing and dev env configuration

### PR DESCRIPTION
**Pull request by triffer**
…oc about assumptions made by the implementation

**Description**

Changes proposed in this pull request:

- Documentation about setting up development environment
- Documentation about configuration for integration tests
- Documentation about assumptions made by the implementations

**Related issue(s)**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -148,6 +194,32 @@ UnDeploy the controller from the cluster:
...
 
+### Configuring integration tests
+The tests can be executed by running the following command:
+
+```sh
+make test
+```
+
+#### IAS stub
+Per default the integration tests use a stub for the IAS API.
```
Comment (muralov) [inline README.md:205]: 👍🏼 it is really important that we document stubbing here.
What do you think that we force integration tests not to run with stubs? I think stub brings little value. Or did misunderstood anything?

File: README.md [hunk 2]
```diff
@@ -148,6 +194,32 @@ UnDeploy the controller from the cluster:
...
+It's also possible to use the real IAS API by setting all the following environment variables:
+
+```sh
+export TEST_EVENTING_AUTH_IAS_URL=https://my-tenant.accounts.ondemand.com
+export TEST_EVENTING_AUTH_IAS_USER=user@sap.com
+export TEST_EVENTING_AUTH_IAS_PASSWORD=password
+```
+
+#### Target Cluster
+Per default the integration tests use a local control plane created by [envtest](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/envtest).
```
Comment (muralov) [inline README.md:215]: 👍🏼

File: README.md [hunk 3]
```diff
@@ -61,6 +62,24 @@ data:
 ## Name reference between resources
 The Kyma CR, which creation is the trigger for the creation of the EventingAuth CR, uses the unique runtime ID of the managed Kyma runtime as name. This name is used as the name for the EventingAuth CR and the IAS application. In this way, the EventingAuth CR and the IAS application can be assigned to the specific managed runtime.
 
+## Assumptions about resource names and structure
```
[the remark this answers is not in the document]
Comment (triffer) [reply README.md:65]: Good point, but I also feel like that `Controller` can be confusing. I'd rephrase this to `Resource Naming Constraints`.

File: README.md (hunk 1 above)
Comment (triffer) [reply README.md:205]: What should we add as documentation for the stubs? I'd not add a lot of details about the stubbing, since this is an implementation detail that may change.
I think using stubs for testing locally and on PRs is better, since the dependency to a 3rd party system might cause flakiness. When using real systems in the e2e tests that always something we should consider.
You can also just start by always using the real IAS app in the CI jobs and then depending on the reliability decide on additional steps. But for normal local development I think the stub should be the default, so IAS credentials are not a requirement during development.

**Review by muralov — APPROVED** [review]
Looks goo
